### DISCUSSION TITLE
Mock name generator options

### DISCUSSION THREAD

**Discussion by jsk95** [ashmind/SourceMock#12] [Ideas]
Name generation of mocks should be either configurable by implementing it or provided with a few options at least

**Comment by ashmind** [comment]
Thanks for the suggestion! Why should it be configurable?

**Reply by jsk95** [reply]
I think we might be assuming people use `I` for interface names, any project that do not use that will have names of their interfaces starting with `I` butchered

**Reply by ashmind** [reply]
I check that second letter is uppercase, so only if it is something like `IXYZ` I suppose.
For this one I might wait for someone to raise this as a problem -- I agree in theory, but rather not do extra complexity in advance.

**Reply by jsk95** [reply]
e.g. IPAddressGenerator -> PAddressGeneratorMock

**Reply by ashmind** [reply]
Good point, I'll add a check that string[2] is lowercase, that should ensure P is meaningful.
